### ISSUE TITLE
Feature request: better way of doing modals

### ISSUE THREAD

**Issue by Deanphillips** [ERROPiX/hydrogen-pack#10]
I'd really love a better way of doing models, that doesn't require doing them all inside a template and allows us to use "#link" to trigger popups. 

Elementors is miles ahead and with things like fluent form, it would be the perfect popup builder

https://github.com/soflyy/oxygen-bugs-and-features/issues/662

**Comment by ERROPiX** [comment]
@Deanphillips do you mean recreate something like https://wordpress.org/plugins/popup-maker/ but use Oxygen to design the content of the popup?

**Comment by Deanphillips** [comment]
Correct! Been asking for something like this for a very long time :)

>

**Comment by Deanphillips** [comment]
David Browne (WPDevDesign) commented on some issues he has with the current modal/way it does things, that it would be amazing to overcome at some point:

- No control over hide animation
- It doesn’t work well with lots of content inside - overflow problems.
- Limited to the click just triggering one thing, the modal reveal.
- Always fixed positioned, not useful for anything absolute.
- It will never be super-smooth due to the way the backdrop is animated from display: none to flex using jQuery. (janky compared to doing it with CSS)
- You can't hide the modal backdrop/make content clickable when you hide it (https://wpdevdesign.com/hide-modal-backdrop-for-notifications-site-messages-in-oxygen/)

### One feature request from me:

1. A way to set cookies (and only show certain modals/notifications based on them)
2. Disable background page scroll when modals fire
